### ISSUE TITLE
Error “format not a string literal and no format arguments” issued when generating R wrapper

### ISSUE THREAD

**Issue by rlaboiss** [swig/swig#2916]
The test suite for R fails for me:

```
$ make check-r-test-suite                                                                                                                                                                                          
checking r test-suite
checking r testcase r_double_delete (with run test)
using C++ compiler: ‘g++ (Debian 13.2.0-25) 13.2.0’
cc1plus: warning: ‘-Werror=’ argument ‘-Werror=implicit-function-declaration’ is not valid for C++
r_double_delete_wrap.cpp: In function ‘void SWIG_R_Raise(SEXP, const char*)’:
r_double_delete_wrap.cpp:1197:11: error: format not a string literal and no format arguments [-Werror=format-security]
 1197 |   Rf_error(Rf_isString(obj) ? CHAR(Rf_asChar(obj)) : msg);
      |   ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: some warnings being treated as errors
make[3]: *** [/usr/lib/R/etc/Makeconf:204: r_double_delete_wrap.o] Error 1
make[3]: Target 'all' not remade because of errors.
make[2]: *** [../../../Examples/Makefile:1382: r_cpp] Error 1
make[1]: *** [Makefile:50: r_double_delete.cpptest] Error 2
[…]
```

This was run on a Debian (trixie) system with the SWIG 4.2.1 tarball, R 4.4.0, and g++ 13.2.0. The option `-Werror=format-security` is included in CFLAGS, as required when building a Debian package.

**Comment by ojwb** [comment]
I think this was already fixed (since the last release) by 3d5157514889c668bc14c245246c388eb23615ea.

**Comment by rlaboiss** [comment]
Indeed, the fix is in Git, but absent from the 4.2.1 release.

Thanks for pointing out the relevant commit. I will integrate it to the Debian package.

I am hereby closing this issue.
